EcommerceStore.Application/Services/OrderService.cs

```diff
@@ -0,0 +1,115 @@
+        {
+            _orderRepository = orderRepository;
+        }
+
+        public async Task CreateOrderAsync(OrderInputModel orderInputModel)
+        {
+            var order = new Order
+            {
+                ModifiedDate = orderInputModel.ModifiedDate,
+                Status = orderInputModel.Status,
+                UserId = orderInputModel.UserId
+            };
+
+            await _orderRepository.CreateAsync(order);
+
+            await _orderRepository.SaveChangesAsync();
+        }
+
+        public async Task<List<OrderViewModel>> GetAllOrdersAsync()
+        {
+            var orders = await _orderRepository.GetAllAsync();
+
+            if (orders is null)
+                throw new ValidationException(NotFoundExceptionMessages.OrderNotFound);
+
+            var ordersViewModel = orders
+                .Where(o => !o.IsDeleted)
+                .Select(o => new OrderViewModel
+                {
+                    Status = o.Status,
+                    CustomerFullName = $"{o.User.FirstName} {o.User.LastName}",
+                    ProductNames = o.ProductOrders
+                        .Where(p => p.OrderId == o.Id)
+                        .Select(p => p.Product.Name)
+                        .ToList(),
+                    TotalPrice = o.ProductOrders
+                        .Where(p => p.OrderId == o.Id)
+                        .Sum(p => p.Price)
+                })
+                .ToList();
+
+            return ordersViewModel;
+        }
+
+        public async Task<OrderViewModel> GetOrderByIdAsync(int orderId)
+        {
+            var order = await _orderRepository.GetByIdAsync(orderId);
+
+            if (order is null || order.IsDeleted)
+                throw new ValidationException(NotFoundExceptionMessages.OrderNotFound, orderId);
+
+            var orderViewModel = new OrderViewModel
+            {
+                Status = order.Status,
+                CustomerFullName = $"{order.User.FirstName} {order.User.LastName}",
+                ProductNames = order.ProductOrders
+                        .Where(p => p.OrderId == order.Id)
+                        .Select(p => p.Product.Name)
+                        .ToList(),
+                TotalPrice = order.ProductOrders
+                        .Where(p => p.OrderId == order.Id)
+                        .Sum(p => p.Price)
+            };
+
+            return orderViewModel;
+        }
+
+        public async Task RemoveOrderAsync(int orderId)
+        {
+            var order = await _orderRepository.GetByIdAsync(orderId);
+
+            if (order is null)
+                throw new ValidationException(NotFoundExceptionMessages.OrderNotFound, orderId);
+
+            order.IsDeleted = true;
+
+            _orderRepository.Update(order);
+
+            await _orderRepository.SaveChangesAsync();
+        }
+
+        public async Task UpdateOrderAsync(int orderId, OrderInputModel orderInputModel)
```

Comment:
let's update only status (if it is failed - then increment back products quantity decremented on create step), otherwise just update status and modified date